plasmapy/diagnostics/charged_particle_radiography.py

```diff
@@ -1115,118 +1263,149 @@ def max_deflection(self):
+
+    vax : `~astropy.units.Quantity` array shape ``(vbins, )``
+        The vertical axis of the synthetic radiograph in meters.
+
+    intensity : `~numpy.ndarray`, shape ``(hbins, vbins)``
+        The number of particles counted in each bin of the histogram.
+    """
 
-            # The factor of 5 here is somewhat arbitrary: we just want a
-            # region a few times bigger than the image of the grid on the
-            # detector, since particles could be deflected out
-            size = 5 * np.array([[-w, w], [-w, w]]) * u.m
+    # results_dict will raise an error if the simulation has not been run.
+    if isinstance(obj, Tracker):
+        d = obj.results_dict
 
-        # Generate the histogram
-        intensity, h, v = np.histogram2d(
-            xloc, yloc, range=size.to(u.m).value, bins=bins
+    # Check if dictioanry-like (dict or npz file)
+    elif isinstance(obj, dict):
+        d = obj
+    else:
+        raise ValueError(
+            "The first argument of synthetic_radiograph must be "
+            "either a cpr.Tracker or an "
+            "output dictionary from cpr.Tracker"
         )
 
-        # h, v are the bin edges: compute the centers to produce arrays
-        # of the right length (then trim off the extra point)
-        h = ((h + np.roll(h, -1)) / 2)[0:-1]
-        v = ((v + np.roll(v, -1)) / 2)[0:-1]
+    if bins is None:
+        bins = [200, 200]
 
-        # Throw a warning if < 50% of the particles are included on the
-        # histogram
-        percentage = np.sum(intensity) / self.nparticles
-        if percentage < 0.5:
-            warnings.warn(
-                f"Only {percentage:.2%} of the particles are shown "
-                "on this synthetic radiograph. Consider increasing "
-                "the size to include more.",
-                RuntimeWarning,
-            )
+    # Note that, at the end of the simulation, all particles were moved
+    # into the image plane.
+
+    # If ignore_grid is True, use the predicted positions in the absence of
+    # simulated fields
+    if ignore_grid:
+        xloc = d["x0"]
+        yloc = d["y0"]
+    else:
+        xloc = d["x"]
+        yloc = d["y"]
+
+    if size is None:
+        # If a detector size is not given, choose a size based on the
+        # particle positions
+        w = np.max([np.max(np.abs(xloc)), np.max(np.abs(yloc))])
+
+        # The factor of 5 here is somewhat arbitrary: we just want a
+        # region a few times bigger than the image of the grid on the
+        # detector, since particles could be deflected out
+        size = 5 * np.array([[-w, w], [-w, w]]) * u.m
+
+    # Generate the histogram
+    intensity, h, v = np.histogram2d(xloc, yloc, range=size.to(u.m).value, bins=bins)
+
+    # h, v are the bin edges: compute the centers to produce arrays
+    # of the right length (then trim off the extra point)
+    h = ((h + np.roll(h, -1)) / 2)[0:-1]
+    v = ((v + np.roll(v, -1)) / 2)[0:-1]
```

Comment:
```suggestion
    h = ((h + np.roll(h, -1)) / 2)[:-1]
    v = ((v + np.roll(v, -1)) / 2)[:-1]
```

Both of these approaches produce the same results with the same execution time.  My mind always trips out when I see a `0` index before the `-1` index, so I have an easier time reading the code I'm suggesting.  However, this is very subjective and doesn't matter in the end so take my suggestion or leave yours in place.  I'll leave the final decision up to you.